browser/src/app/TilesMiddleware.ts

```diff
@@ -538,9 +424,12 @@ class TileManager {
 				},
 				this.pendingDeltas.map((x: any) => x.rawDelta.buffer),
 			);
-			++this.pendingTransactions;
+			this.pendingDeltas.length = 0;
 		} else {
-			for (var e of this.pendingDeltas) {
+			const bitmaps = [];
+			const pendingDeltas = this.pendingDeltas;
+			this.pendingDeltas = [];
+			for (var e of pendingDeltas) {
 				// Synchronous path
 				var tile = this.tiles[e.key];
```

Comment:
I guess we need a if(!tile) check here, like the one for onWorkerMessage? If we don't put a check here, and if tile may not be present, then below line will break the order:
tile.image = bitmaps.shift();

Reply:
It certainly doesn't hurt - I'm not sure we do on the synchronous path as tiles are generally only removed on garbage collection, which only happens after a transaction has been handled. But nested transactions are a possibility and I can't envision every way this code will be triggered - plus you're right, this should be symmetric regardless :)